### PR TITLE
[ingress-nginx] add ip and hostname to ingress-nginx-controller status

### LINKED ISSUE #7718: [ingress-nginx] Show IP in controller status
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/deckhouse/deckhouse/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/deckhouse/deckhouse/issues) for an issue that matches the one I want to file, without success.

### Use case. Why is this important?

-

### Proposed Solution

1. Add status field for [IngressNginxController CRD](https://github.com/deckhouse/deckhouse/blob/main/modules/402-ingress-nginx/crds/ingress-nginx.yaml)
Something like:
```yaml
status:  
  loadBalancer:
    address: 1.2.3.4
    id: lb-abracadabra012355.aws.whatever.here
```
2. Add it to the AdditionalPrinterColumns
3. Create a hook, subscribe to a service for LB
4. Fetch an IP and pass it to the CRD Status

### Additional Information

Only for inlets `LoadBalancer` and `LoadBalancerWithProxyProtocol`

### PR DESCRIPTION
**Pull request by ipaqsa**
## Description
Provides an additional fields for ingress-nginx-controller status. These fields represent assigned ip and hostname of the external loadBalancer.

## Why do we need it, and what problem does it solve?
Closes #7718

## What is the expected result?
The ip and hostname are displayed when getting ingress-nginx-controller with LoadBalancer inlet. 

## Checklist
- [ ] The code is covered by unit tests.
- [ ] e2e tests passed.
- [ ] Documentation updated according to the changes.
- [ ] Changes were tested in the Kubernetes cluster manually.

## Changelog entries
```changes
section: ingress-nginx
type: feature
summary: add ip and hostname to ingress-nginx-controller status
```

### CODE REVIEW

File: modules/402-ingress-nginx/hooks/update_ingress_address.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+*/
+
+package hooks
+
+import (
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/sdk"
+	"github.com/flant/shell-operator/pkg/kube/object_patch"
+	"github.com/flant/shell-operator/pkg/kube_events_manager/types"
+	v12 "k8s.io/api/core/v1"
```
Comment (yalosev) [inline modules/402-ingress-nginx/hooks/update_ingress_address.go:24]: corev1

File: modules/402-ingress-nginx/hooks/update_ingress_address.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+type loadBalancerService struct {
+	name     string
+	hostname string
+	ip       string
+}
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	OnAfterHelm: &go_hook.OrderedConfig{Order: 10},
+	Queue:       "/modules/ingress-nginx",
```
Comment (yalosev) [inline modules/402-ingress-nginx/hooks/update_ingress_address.go:38]:
```suggestion
	Queue:       "/modules/ingress-nginx/service-discovery-ip",
```

File: modules/402-ingress-nginx/crds/doc-ru-ingress-nginx.yaml [hunk 3]
```diff
@@ -4,6 +4,16 @@ spec:
       schema: &schema
         openAPIV3Schema:
           properties:
+            status:
+              properties:
+                loadBalancer:
+                  properties:
+                    ip:
+                      description: |
+                        Выделенный внешний IP адрес балансировщика нагрузки.
```
Comment (z9r5) [inline modules/402-ingress-nginx/crds/doc-ru-ingress-nginx.yaml:13]:
```suggestion
                        Внешний IP-адрес балансировщика нагрузки.
```

File: modules/402-ingress-nginx/crds/doc-ru-ingress-nginx.yaml [hunk 4]
```diff
@@ -4,6 +4,16 @@ spec:
...
+                    hostname:
+                      description: |
+                        Выделенный внешний хостнейм балансировщика нагрузки.
```
Comment (z9r5) [inline modules/402-ingress-nginx/crds/doc-ru-ingress-nginx.yaml:16]:
```suggestion
                      description: |
                        IP-адрес балансировщика нагрузки.
                    hostname:
                      description: |
                        DNS-имя балансировщика нагрузки.
```

File: modules/402-ingress-nginx/crds/ingress-nginx.yaml [hunk 5]
```diff
@@ -551,5 +551,558 @@ spec:
...
+            status:
+              type: object
+              properties:
+                loadBalancer:
+                  type: object
+                  properties:
+                    ip:
+                      type: string
+                      description: |
+                        External IP address of external provider LoadBalancer
```
Comment (z9r5) [inline modules/402-ingress-nginx/crds/ingress-nginx.yaml:570]:
```suggestion
                        The IP address of the load balancer.
```

File: modules/402-ingress-nginx/crds/ingress-nginx.yaml [hunk 6]
```diff
@@ -551,5 +551,558 @@ spec:
...
+                    hostname:
+                      type: string
+                      description: |
+                        External hostname of external provider LoadBalancer
```
Comment (z9r5) [inline modules/402-ingress-nginx/crds/ingress-nginx.yaml:574]:
```suggestion
                        The hostname of the LoadBalancer.
```

**Review by z9r5 — COMMENTED** [review]
@yalosev Подумалось что внешний и выделенный тут избыточно. Адрес может быть внутренний же в каком-то случае (от конфигурации провайдера зависит), ну а то что он выделнный и так понятно - это же поле статуса.  
?

**Review by z9r5 — APPROVED** [review]

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @miklezzzz. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Comment by miklezzzz** [comment]
image

**Comment by deckhouse-BOaTswain** [comment]
Aye, aye, @miklezzzz. I've started the workflow for label 'e2e/run/yandex-cloud'...

**Review by yalosev — APPROVED** [review]

**Pull request merged**